src/char_array.c

```diff
@@ -139,6 +139,12 @@ rcutils_char_array_expand_as_needed(rcutils_char_array_t * char_array, size_t ne
     return RCUTILS_RET_OK;
   }
 
+  // Make sure we expand by at least 1.5x the old capacity
```

Comment:
I think it would be better to move this to header file doc section, not in the implementation.

Reply:
The thing is, the change to 1.5x is an implementation detail, not something the user really needs to worry about.  That is, other than a slight increase in memory, the user won't really notice anything that changes because of this.  It should just avoid the reallocation logic on every small increase to the size.  Because of that, I prefer to leave it in the implementation.